tastebuddies/models.py

```diff
@@ -0,0 +1,178 @@
+    confirmed = Column(Boolean, default=False)
+    ver_code = Column(Integer)
+    restaurants = Column(Text)
+    food = Column(Text)
+    criteria_id = Column(Integer, ForeignKey('criteria.id'))
+    admin_groups = relationship("Group", backref='admin')
+
+    @validates('email')
+    def validate_email(self, key, email):
+        try:
+            assert '@' in email
+            assert '.' in email
+            return email
+        except:
+            raise TypeError('Please enter a vaild email address')
+
+    @property
+    def __acl__(self):
+        acl = []
+        acl.append((Allow, self.username, 'owner'))
+        for group in self.groups:
+            acl.append((Allow, 'group:{}'.format(group.id), 'connect'))
+        return acl
+
+    def __repr__(self):
+        return "<User({} {}, username={})>".format(self.firstname,
+                                                   self.lastname,
+                                                   self.username)
+
+
+class Criteria(Base, TableSetup):
+    """Constant variables are used to generate options for the user
+    to choose from on the front end.  Columns are PickleTypes and expect
+    to receive and return lists.  Each user has one set of criteria and
+    each group has one set of criteria and vice versa for a one to one
+    relationship.
+    """
+    __tablename__ = 'criteria'
+    TASTES = ['Sour', 'Sweet', 'Salty', 'Spicy', 'Bitter', 'Italian',
+              'Chinese', 'American Classic', 'German', 'British',
+              'French', 'Vietnamese', 'Japanese', 'Pub', 'Persian',
+              'Mediterranian', 'Greek', 'Afghan', 'Somolian',
+              'Thai', 'Barbecue', 'Soul', 'Ethiopian', 'Jamaican',
+              'Mexican', 'Korean',
+              ]
+    DIETS = ['Vegetarian', 'Vegan', 'Gluten Free', 'Low Carb']
+    LOCATIONS = ['Seattle', 'Kitsap', 'Eastside', 'Skagit', 'South King']
+    AGES = ['18-24', '25-34', '35-44', '45-54', '55-64', '65-74', '75+']
+    COSTS = ['$', '$$', '$$$', '$$$$']
+    taste = Column(PickleType)
+    age = Column(PickleType)
+    location = Column(PickleType)
+    cost = Column(PickleType)
+    diet = Column(PickleType)
+    group = relationship('Group', uselist=False, backref='criteria')
+    user = relationship('User', uselist=False, backref='criteria')
+
+    def __repr__(self):
+        if self.user:
+            return"<Criteria for %s>" % (self.user.username)
+        else:
+            return"<Criteria for %s>" % (self.group.name)
+
+
+class Group(Base, TableSetup):
+    """We expect forum to be input as an ordered dictionary where keys
+    represent titles and values are a list of posts.  Many to many
+    realationship with users, one to one relationship with criteria
+    and one to many relationship with users as admins.
+    """
+    __tablename__ = 'groups'
+    name = Column(Text, unique=True, nullable=False)
+    description = Column(Text, nullable=False)
+    forum = Column(PickleType)
```

Comment:
I am suspicious of this `PickleType`.  It's almost certain that there's a better, more idiomatic way to do whatever you're trying to do here.